Classes/Domain/Repository/CommentRepository.php

```diff
@@ -246,12 +246,23 @@ protected function getFindByEmailAndPostIdConstraints(QueryInterface $query, $em
      *
      * @return object
      */
-    protected function getValidConstraints(QueryInterface $query)
+    protected function getValidConstraints(QueryInterface $query, $excludeHiddenPosts = false)
     {
         $constraints = $query->logicalAnd(
             $query->equals('spam', 0),
             $query->equals('approved', 1)
         );
+        // By default, comments of hidden posts will be included in the result, because there is no relation / join from comments to posts
```

Comment:
> What is the use case for this

Yes, basically every view we are handling with comments which could have hidden posts. In BE pretty much everywhere. FE is fine when comments are not available with for hidden posts.

> Couldn't this be fixed by temporarily changing the enablefields for the posts table? Could you give this a try?

Probably. Sounds bad, but there is already something similar :-/
 https://github.com/fnagel/t3extblog/blob/master/Classes/Domain/Repository/PostRepository.php#L83

Another issue is how this would behave when using a multi language setup.